standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -3034,6 +3040,10 @@ private boolean drop_table_core(final RawStore ms, final String catName, final S
         deletePartitionData(partPaths, ifPurge, ReplChangeManager.shouldEnableCm(db, tbl));
         // Delete the data in the table
         deleteTableData(tblPath, ifPurge, ReplChangeManager.shouldEnableCm(db, tbl));
+      } else if (TxnUtils.isTransactionalTable(tbl)) {
```

Comment:
Why compact when dropping?

Reply:
we are not dropping, we are delegating cleanup to the Cleaner do remove when possible